snappi/snappicommon.py

```diff
@@ -159,6 +274,7 @@ def _encode(self):
         """Helper method for serialization
         """
         output = {}
+        self.validate()
         for key, value in self._properties.items():
```

Comment:
do not `output[key] = value` if value is `None`. This would prevent "key: null" from appearing in serialized output.